tests/test_setup_prometheus_adapter_config.py

```diff
@@ -0,0 +1,159 @@
+    create_instance_uwsgi_scaling_rule,
+)
+from paasta_tools.setup_prometheus_adapter_config import (
+    create_prometheus_adapter_config,
+)
+from paasta_tools.setup_prometheus_adapter_config import (
+    should_create_uwsgi_scaling_rule,
+)
+
+
+@pytest.mark.parametrize(
+    "instance_name,autoscaling_config,expected",
+    [
+        (
+            "not_uwsgi_autoscaled",
+            {
+                "metrics_provider": "mesos_cpu",
+                "decision_policy": "bespoke",
+                "moving_average_window_seconds": 123,
+                "setpoint": 0.653,
+            },
+            False,
+        ),
+        (
+            "uwsgi_autoscaled_no_prometheus",
+            {
+                "metrics_provider": "uwsgi",
+                "moving_average_window_seconds": 124,
+                "setpoint": 0.425,
+            },
+            False,
+        ),
+        (
+            "uwsgi_autoscaled_prometheus",
+            {
+                "metrics_provider": "uwsgi",
+                "use_prometheus": True,
+                "moving_average_window_seconds": 544,
+                "setpoint": 0.764,
+            },
+            True,
+        ),
+    ],
+)
+def test_should_create_uswgi_scaling_rule(
+    instance_name: str, autoscaling_config: AutoscalingParamsDict, expected: bool
+) -> None:
+    should_create, reason = should_create_uwsgi_scaling_rule(
+        instance=instance_name, autoscaling_config=autoscaling_config
+    )
+
+    assert should_create == expected
+    if expected:
+        assert reason is None
+    else:
+        assert reason is not None
+
+
+def test_create_instance_uwsgi_scaling_rule() -> None:
+    service_name = "test_service"
+    instance_name = "test_instance"
+    paasta_cluster = "test_cluster"
+    autoscaling_config: AutoscalingParamsDict = {
+        "metrics_provider": "uwsgi",
+        "setpoint": 0.1234567890,
+        "moving_average_window_seconds": 20120302,
+        "use_prometheus": True,
+    }
+
+    rule = create_instance_uwsgi_scaling_rule(
+        service=service_name,
+        instance=instance_name,
+        paasta_cluster=paasta_cluster,
+        autoscaling_config=autoscaling_config,
+    )
+
+    # we test that the format of the dictionary is as expected with mypy
+    # and we don't want to test the full contents of the retval since then
+    # we're basically just writting a change-detector test - instead, we test
+    # that we're actually using our inputs
+    assert service_name in rule["seriesQuery"]
+    assert instance_name in rule["seriesQuery"]
+    assert paasta_cluster in rule["seriesQuery"]
+    # these two numbers are distinctive and unlikely to be used as constants
+    assert str(autoscaling_config["setpoint"]) in rule["metricsQuery"]
+    assert (
+        str(autoscaling_config["moving_average_window_seconds"]) in rule["metricsQuery"]
+    )
+
+
+def test_create_prometheus_adapter_config(tmpdir: LocalPath) -> None:
```

Comment:
Is this coming from a fixture? I don't see it defined anywhere.

Most of our other paasta unit tests would probably mock.patch `get_services_for_cluster` and `config_loader.instance_configs`. Might be better to do that here, to keep this unit test slightly purer, or to move this to general_itests or k8s_itests.

Reply:
https://github.com/Yelp/paasta/pull/3027/commits/6bb1645ad7ad231e7f865aae4c8849ab45f986e1 refactors this to be more pure